File: pubnub-gson/pubnub-gson-docs/src/main/java/com/pubnub/docs/publishAndSubscribe/fire/FireOthers.java
package com.pubnub.docs.publishAndSubscribe.fire;

import com.pubnub.api.PubNubException;
import com.pubnub.api.UserId;
import com.pubnub.api.java.PubNub;
import com.pubnub.api.java.v2.PNConfiguration;
import com.pubnub.api.java.v2.entities.Channel;
import com.pubnub.api.models.consumer.PNPublishResult;

import java.util.Arrays;

public class FireOthers {
    private void fireMessage() throws PubNubException {
        // https://www.pubnub.com/docs/sdks/java/api-reference/publish-and-subscribe#fire-a-message-to-a-channel

        // snippet.fireMessage
        PNConfiguration.Builder configBuilder = PNConfiguration.builder(new UserId("yourUserId"), "demo");
        configBuilder.publishKey("demo");

        PNConfiguration pnConfiguration = configBuilder.build();
        PubNub pubnub = PubNub.create(pnConfiguration);
        Channel channel = pubnub.channel("myChannel");

        channel.fire(Arrays.asList("hello", "there"))
                .usePOST(true)
                .async(result -> {
                    result.onSuccess((PNPublishResult publishResult) -> {
                        System.out.println("publish worked! timetoken: " + publishResult.getTimetoken());
                    }).onFailure( (PubNubException exception) -> {
                        System.out.println("error happened while publishing: " + exception.toString());
                    });
                });
        // snippet.end
    }

}

File: pubnub-kotlin/pubnub-kotlin-docs/src/main/kotlin/com/pubnub/docs/publishAndSubscribe/fire/FireOthers.kt
package com.pubnub.docs.publishAndSubscribe.fire

import com.google.gson.JsonObject
import com.pubnub.api.PubNub
import com.pubnub.api.UserId
import com.pubnub.api.models.consumer.PNPublishResult

class FireOthers {

    private fun fireBasic() {
        // https://www.pubnub.com/docs/sdks/kotlin/api-reference/publish-and-subscribe#basic-usage-1

        // snippet.fireBasic
        val configBuilder = com.pubnub.api.v2.PNConfiguration.builder(UserId("myUserId"), "demo").apply {
            publishKey = "demo"
        }

        val pubnub = PubNub.create(configBuilder.build())
        val channel = pubnub.channel("myChannel")
        val myMessage = JsonObject().apply {
            addProperty("text", "Hello, world")
        }

        channel.fire(myMessage).async { result ->
            result.onFailure { exception ->
                println("Error while publishing")
                exception.printStackTrace()
            }.onSuccess { fireResult ->
                println("Message sent, timetoken: ${fireResult.timetoken}")
            }
        }

        // snippet.end
    }

}